pkg/services/sqlstore/database_wrapper.go

```diff
@@ -100,12 +99,15 @@ func (h *databaseQueryWrapper) OnError(ctx context.Context, err error, query str
 	return err
 }
 
-type databaseQueryWrapperParser struct {
+// databaseQueryWrapperDriver satisfies the xorm.io/core.Driver interface
+type databaseQueryWrapperDriver struct {
 	dbType string
 }
 
-func (hp *databaseQueryWrapperParser) Parse(string, string) (*core.Uri, error) {
-	return &core.Uri{
-		DbType: core.DbType(hp.dbType),
-	}, nil
+func (hp *databaseQueryWrapperDriver) Parse(driverName, dataSourceName string) (*core.Uri, error) {
+	driver := core.QueryDriver(hp.dbType)
+	if driver == nil {
+		return nil, fmt.Errorf("could not find driver with name %s", hp.dbType)
+	}
+	return core.QueryDriver(hp.dbType).Parse(driverName, dataSourceName)
```

Comment:
Isn't this equivalent to the following? I think being explicit about that would help in reading the code (looks as if the method can work with a `nil` receiver).

```suggestion
	return driver.Parse(driverName, dataSourceName)
```

Reply:
Yes, you're right. Thank you!